cmake/bintools/oneApi/target.cmake

```diff
@@ -1,12 +1,23 @@
 # SPDX-License-Identifier: Apache-2.0
 
 if(DEFINED TOOLCHAIN_HOME)
-  set(find_program_clang_args PATHS ${TOOLCHAIN_HOME} ${ONEAPI_PYTHON_PATH} NO_DEFAULT_PATH)
+  set(find_program_clang_args PATHS ${TOOLCHAIN_HOME} ${ONEAPI_LLVM_BIN_PATH} NO_DEFAULT_PATH)
   set(find_program_binutils_args PATHS ${TOOLCHAIN_HOME} )
 endif()
 
+execute_process(COMMAND ${CMAKE_C_COMPILER} --version OUTPUT_VARIABLE ONEAPI_VERSION)
+string(REGEX REPLACE "[^0-9]*([0-9.]+) .*" "\\1" ONEAPI_VERSION ${ONEAPI_VERSION})
+
+if(ONEAPI_VERSION VERSION_LESS "2023.0.0")
+  message(FATAL_ERROR "Unsupported version of the toolchain - need to use 2023.0.0 or newer")
+endif()
+
```

Comment:
This could go to a `FindoneApi.cmake` file in `cmake/modules`, which would allow us to do a clean `find_package(oneApi 2023.0.0)` call instead.

That would cleanup this code and make error messaging and similar aligned with other version checking.
Also it makes it clearer where to update version to use.

It would also be possible to let the `FindoneApi.cmake` code set all the oneApi relevant programs as CMake variables.

For inspiration, take a look at `FindDtc.cmake` and how it's used:
https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/modules/FindDtc.cmake
https://github.com/zephyrproject-rtos/zephyr/pull/54075/files#diff-76066c3ade0054d80b41edd6ef022e28f99b3daf40756ab6889cf0cf3914206fR8-R14

Reply:
Added a FindoneApi.cmake